✨Feature: 트렌드 미션 좋아요 기능 구현

Ko-udon opened the pull request:
## 개요
- 수정사항
  - 테스트 코드
    - 기존의 사용자 생성 코드를 수정하였습니다.
  - accounts > models.py
    - 순환 참조 방지를 위해 import 선언문 수정
  - accounts > serializers.py
    - LikeSerializer 정의

- 구현기능
  - 해당하는 트렌드 미션에 좋아요를 등록하는 기능 구현
  - 이미 좋아요 등록이 되어있다면, 좋아요 취소
  - 다시 등록 요청시 좋아요 등록
  - 새로운 리소스를 생성, 삭제, 생성 반복하기 때문에 put 요청으로 구현

- 비고
  - 테스트 수행
  - 좋아요 실패 테스트는 성공했지만 좋아요 등록은 실패
  - 테스트 코드에서 사용자 정보를 토큰에 담아서 보내야됨

Resolves: #34

## PR 유형
- [x] 새로운 기능 추가
- [x] 코드 리팩토링
- [x] 테스트 추가, 테스트 리팩토링


## PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.  
- [x] 변경 사항에 대한 테스트를 했습니다.

Ko-udon committed:
♻️ Refactor: 테스트 코드 수정 및 accounts model, serializer 수정

- 수정사항
  - 테스트 코드
    - 기존의 사용자 생성 코드를 수정하였습니다.
  - accounts > models.py
    - 순환 참조 방지를 위해 import 선언문 수정
  - accounts > serializers.py
    - LikeSerializer 정의

Ko-udon committed:
✨ Feature: 트렌드 미션 좋아요 API

- 구현기능
  - 해당하는 트렌드 미션에 좋아요를 등록하는 기능 구현
  - 이미 좋아요 등록이 되어있다면, 좋아요 취소
  - 다시 등록 요청시 좋아요 등록
  - 새로운 리소스를 생성, 삭제, 생성 반복하기 때문에 put 요청으로 구현

- 비고
  - 테스트 수행
  - 좋아요 실패 테스트는 성공했지만 좋아요 등록은 실패
  - 테스트 코드에서 사용자 정보를 토큰에 담아서 보내야됨

nayeongdev commented on accounts/models.py:
```diff
@@ -115,8 +115,12 @@ class Like(models.Model):
     )
+
+    # 순환 참조 방지를 위해 아래와 같이 import
+    from trend_missions.models import TrendMission
```
import문은 보통 최상단에서 선언하던데, 여기에다 쓰신 이유는 무엇인가요??

Ko-udon replied:
순환 참조가 발생해서 그렇습니다.
accounts model에서 TrendMission을 참조하고, TrendMission model에서 accounts를 참조해서 발생하였습니다.

순환 참조를 막기 위해 부득이하게 모델 정의 안에다가 import를 하였습니다.

hantang820 reviewed (approved):
확인했습니다👍

Ko-udon committed:
♻️ Refactor: 트렌드 미션 좋아요 코드 수정

- 변경사항
  - 토큰에서 사용자를 판별하지 않고 url에 사용자 정보를 담아 요청하도록 수정

- 비고
  - 테스트 코드 수정
  - 테스트 수행 완료
